src/main/java/io/candydoc/ddd/extract_ddd_concepts/DDDConceptsExtractionService.java

```diff
@@ -64,72 +70,125 @@ private void trackAndApply(List<Event> occurredEvents) {
     log.info("Check concept interactions from {}", command.getConceptName());
     trackAndApply(interactionChecker.check(command));
   }
 
-  public void handle(ExtractAggregates command) {
-    log.info("Extract aggregates from {}", command.getPackageToScan());
-    trackAndApply(aggregatesExtractor.extract(command));
+  public void apply(AggregateFound event) {
+    this.handle(CheckConceptInteractions.builder().conceptName(event.getCanonicalName()).build());
   }
 
   public void apply(BoundedContextFound event) {
+    this.handle(ExtractAggregates.builder().packageToScan(event.getPackageName()).build());
+    this.handle(ExtractBoundedContexts.builder().packageToScan(event.getPackageName()).build());
     this.handle(ExtractCoreConcepts.builder().packageToScan(event.getPackageName()).build());
-    this.handle(ExtractValueObjects.builder().packageToScan(event.getPackageName()).build());
-    this.handle(ExtractDomainEvents.builder().packageToScan(event.getPackageName()).build());
     this.handle(ExtractDomainCommands.builder().packageToScan(event.getPackageName()).build());
-    this.handle(ExtractAggregates.builder().packageToScan(event.getPackageName()).build());
+    this.handle(ExtractDomainEvents.builder().packageToScan(event.getPackageName()).build());
+    this.handle(ExtractValueObjects.builder().packageToScan(event.getPackageName()).build());
   }
 
   public void apply(CoreConceptFound event) {
     this.handle(CheckConceptInteractions.builder().conceptName(event.getCanonicalName()).build());
   }
 
-  public void apply(InteractionBetweenConceptFound event) {}
-
-  public void apply(ValueObjectFound event) {
+  public void apply(DomainCommandFound event) {
     this.handle(CheckConceptInteractions.builder().conceptName(event.getCanonicalName()).build());
   }
 
   public void apply(DomainEventFound event) {
     this.handle(CheckConceptInteractions.builder().conceptName(event.getCanonicalName()).build());
   }
 
-  public void apply(DomainCommandFound event) {
-    this.handle(CheckConceptInteractions.builder().conceptName(event.getCanonicalName()).build());
+  public void apply(SharedKernelFound event) {
+    this.handle(ExtractCoreConcepts.builder().packageToScan(event.getPackageName()).build());
+    this.handle(ExtractValueObjects.builder().packageToScan(event.getPackageName()).build());
+    this.handle(ExtractSharedKernels.builder().packageToScan(event.getPackageName()).build());
   }
 
-  public void apply(AggregateFound event) {
+  public void apply(ValueObjectFound event) {
     this.handle(CheckConceptInteractions.builder().conceptName(event.getCanonicalName()).build());
   }
 
+  public void apply(ConceptRuleViolated event) {}
+
+  public void apply(InteractionBetweenConceptFound event) {}
+
   public void apply(NameConflictBetweenCoreConcepts event) {}
 
-  public void apply(ConceptRuleViolated event) {}
+  private void checkNoInnerBoundedContextNorSharedKernelInside(List<Event> events) {
```

Comment:
No way. The verification is done by checking interactions between concepts. Bounded contexts and Shared Kernels are special cases: we need to check they're contained inside others. It should be a specific command when a BoundedContext or a SharedKernel is found that leads to an interaction checker.